x/crosschain/keeper/keeper_cross_chain_tx.go

```diff
@@ -157,16 +158,39 @@ func (k Keeper) CctxAllPending(c context.Context, req *types.QueryAllCctxPending
 		return nil, status.Error(codes.InvalidArgument, "invalid request")
 	}
 	ctx := sdk.UnwrapSDKContext(c)
+	if req.ChainId > goMath.MaxInt64 {
+		return nil, status.Error(codes.InvalidArgument, "invalid chain id")
+	}
+	pendingCrossChainTxList, err := k.GetActualPendingCCTX(ctx, int64(req.ChainId))
+	if err != nil {
+		return nil, err
+	}
+	return &types.QueryAllCctxPendingResponse{CrossChainTx: pendingCrossChainTxList}, nil
+}
+
+func (k Keeper) CctxAllPendingCount(c context.Context, _ *types.QueryAllCctxPendingCountRequest) (*types.QueryAllCctxPendingCountResponse, error) {
```

Comment:
this separate RPC call seems a bit redunance to me: 
on one hand, the nonceHigh-nonceLow gives the same result here per chain (which is better than lumped all pendings count together, and much cheaper)

If really need the count, a better place would be to use CctxAllPending and have it with proper pagination (the result will include a total count). This is the same cost as here as all the pending cctxs are collected.

Reply:
The difference between high an low would not include an cctxs which might be have nonces less tha nonce low . As  described on the comment lines.
That is the sole reason for creating this pr , if those ctxs are not needed we can go ahead and close the pr and continue using getpending nonces

As discussed with front end they jsut need the count and not the actual cctx , keeping the same cost this would avoid the need for them to query multiple pages when trying to get just the count .

Reply:
The difference between high an low would not include an cctxs which might be have nonces less tha nonce low . As described on the comment lines.
That is the sole reason for creating this pr , if those ctxs are not needed we can go ahead and close the pr and continue using getpending nonces

As discussed with front end they jsut need the count and not the actual cctx , keeping the same cost this would avoid the need for them to query multiple pages when trying to get just the count .

I can however improve this query by avoiding the second loop in getActualPendimg

Reply:
see this http://46.4.15.110:1317/zeta-chain/crosschain/outTxTracker

at the end: 
<img width="366" alt="image" src="https://github.com/zeta-chain/node/assets/88689859/380a634a-09ef-46c8-9178-ae3878c82504">

This is the standard behavior of pagination; it returns a total count regardless of how many records is returned in this page. 

If you just need the count, query with page limit=1, and get the count. 
Isn't this better than a separate RPC just for count?

Reply:
Here's an example of outTxTracker: 
http://46.4.15.110:1317/zeta-chain/crosschain/outTxTracker?pagination.limit=1

Reply:
hmm... the total is 0 in this case. Wut??!

@lumtis do you know whether this is expected behavior of pagination?